src/optim/FishLeg/fishleg_likelihood.py

```diff
@@ -107,18 +158,11 @@ def __init__(self, device: str = "cpu") -> None:
         self.device = device
 
     def nll(self, observations: torch.Tensor, preds: torch.Tensor) -> torch.Tensor:
-
-        result = (
-            -torch.sum(
-                torch.log(1 - preds + 1e-5) * (1.0 - observations)
-                + torch.log(preds + 1e-5) * observations
-            )
-            / preds.shape[0]
-        )
-        return result
+        bce = torch.sum(preds * (1. - observations) + torch.nn.Softplus()(-preds))
+        return bce / preds.shape[0]
```

Comment:
Please can we check this refactor of nll? Further to our conversation on Slack, this may be causing an issue?

Also the method below where probs is substituted for logits.